src/lua/LuaSpace.cpp

```diff
@@ -298,6 +319,62 @@ static int l_space_put_ship_on_route(lua_State *l)
 	return 0;
 }
 
+/*
+ * Method: PutShipIntoOrbit
+ *
+ * Puts ship into orbit of target planet, moon or star
+ *
+ * > Space.PutShipIntoOrbit(ship, target)
+ *
+ * Parameters:
+ *
+ *   ship - a <Ship> object to be moved
+ *
+ *   target - the <Star> or <Planet> to orbit
+ *
+ * Availability:
+ *
+ *  October 2023
+ *
+ * Status:
+ *
+ *  experimental
+ */
+static int l_put_ship_into_orbit(lua_State* l)
+{
+	Ship *s = LuaObject<Ship>::CheckFromLua(1);
+	Body *b = LuaObject<Body>::CheckFromLua(2);
+	const SystemBody* sbody = b->GetSystemBody();
+	if (!sbody)
+	{
+		return luaL_error(l, "the target body doesn't have a system body");
+	}
+	if (!sbody->IsPlanet() && !sbody->IsMoon() && sbody->GetSuperType() != SystemBodyType::SUPERTYPE_STAR)
+	{
+		return luaL_error(l, "the target body is not a planet, moon, or star");
+	}
```

Comment:
@Max5377 looked at your correction, it seems worth tweaking a little more:
```diff
-	if (!sbody->GetMass() && !b->GetPhysRadius()) {
-		return luaL_error(l, "the target body has zero mass and physical radius");
+	if (!sbody->GetMass() || !b->GetPhysRadius()) {
+		return luaL_error(l, "the target body has zero mass or zero physical radius");
```
Otherwise, for example, a body with zero mass but non-zero radius (an orbital) will not be discarded. (If I'm not mistaken and the orbital has zero mass, otherwise we'll have to do some more reworking.)

Reply:
I'd recommend separating this into two separate `if` blocks, testing individually for `sbody->GetMass() == 0` and `sbody->GetPhysRadius() == 0`. Both parameters are required to be non-zero to generate a valid orbit.

Reply:
@Web-eWorks Did you approach with a little tweaking by borrowing physical radius from body itself, because SystemBody doesn't have physical radius.